kasa/tests/fakeprotocol_smartcamera.py

```diff
@@ -0,0 +1,213 @@
+                        fixture_info.data, fixture_info.name
+                    )
+                else:
+                    warn(
+                        f"Could not find child fixture {child_fixture_name}",
+                        stacklevel=1,
+                    )
+            else:
+                pass
+                # TODO: load camera protocol
+        # Replace child infos with the infos that found child fixtures
+        if child_infos:
+            self.info["getChildDeviceList"]["child_device_list"] = (
+                found_child_fixture_infos
+            )
+        return child_protocols
+
+    async def _handle_control_child(self, params: dict):
+        """Handle control_child command."""
+        device_id = params.get("device_id")
+        assert device_id in self.child_protocols, "Fixture does not have child info"
+
+        child_protocol: SmartProtocol = self.child_protocols[device_id]
+
+        request_data = params.get("request_data", {})
+
+        child_method = request_data.get("method")
+        child_params = request_data.get("params")  # noqa: F841
+
+        resp = await child_protocol.query({child_method: child_params})
+        resp["error_code"] = 0
+        for val in resp.values():
+            return {
+                "result": {"response_data": {"result": val, "error_code": 0}},
+                "error_code": 0,
+            }
+
+    @staticmethod
+    def _get_param_set_value(info: dict, set_keys: list[str], value):
+        for key in set_keys[:-1]:
+            info = info[key]
+        info[set_keys[-1]] = value
+
+    SETTERS = {
+        ("system", "sys", "dev_alias"): [
+            "getDeviceInfo",
+            "device_info",
+            "basic_info",
+            "device_alias",
+        ],
+        ("lens_mask", "lens_mask_info", "enabled"): [
+            "getLensMaskConfig",
+            "lens_mask",
+            "lens_mask_info",
+            "enabled",
+        ],
+    }
+
+    async def _send_request(self, request_dict: dict):
+        method = request_dict["method"]
+
+        info = self.info
+        if method == "controlChild":
+            return await self._handle_control_child(
+                request_dict["params"]["childControl"]
+            )
+
+        if method == "set":
+            for key, val in request_dict.items():
+                if key != "method":
+                    module = key
+                    section = next(iter(val))
+                    skey_val = val[section]
+                    for skey, sval in skey_val.items():
+                        section_key = skey
+                        section_value = sval
+                    break
+            if setter_keys := self.SETTERS.get((module, section, section_key)):
+                self._get_param_set_value(info, setter_keys, section_value)
+                return {"error_code": 0}
+            else:
+                return {"error_code": -1}
+        elif method[:3] == "get":
```

Comment:
The one for `set` compares to the whole method, is that correct?

Reply:
Yes, the get can do a generic lookup whereas the set needs to be configured for each setter